src/main/java/com/diffmin/patch/PatchApplication.java

```diff
@@ -77,6 +77,28 @@ private static void performInsertion(
                                 ((CtExecutable<?>) toBeInserted.getParent()).getThrownTypes());
                 ((CtExecutable<?>) inWhichElement).setThrownTypes(thrownTypesCopy);
                 break;
+            case CONTAINED_TYPE:
+                // Inserting into CtPackage
+                Set<CtType<?>> typesCopy =
+                        new HashSet<>(((CtPackage) toBeInserted.getParent()).getTypes());
+                for (CtType<?> type : typesCopy) {
+                    type.setParent(inWhichElement);
+                }
+                ((CtPackage) inWhichElement).setTypes(typesCopy);
+
+                // Inserting into CtCompilationUnit
+                CtCompilationUnit cu =
+                        inWhichElement
+                                .getFactory()
+                                .CompilationUnit()
+                                .getOrCreate((CtPackage) inWhichElement);
```

Comment:
This is not the right compilation unit. You're getting the compilation unit of the package (which is a `package-info.java` file). What you want is the compilation unit of the type.

Reply:
Is getting the compilation unit of the type _more_ correct? The current implementation works for the test case. Maybe because the package in our case will contain only one two files - `package.info` and a `PREV` file.

Reply:
I will have to give more thought to this because there can be a case when the previous model is an empty file so I cannot just `findFirst` type and get its compilation unit.